src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -164,13 +219,20 @@ public void Execute(ApplicationUpdateCommand message)
 
             if (latestAvailable != null)
             {
-                InstallUpdate(latestAvailable);
+                ExecuteInstallUpdate(message, latestAvailable);
             }
         }
 
         public void Execute(InstallUpdateCommand message)
         {
-            InstallUpdate(message.UpdatePackage);
+            var latestAvailable = _checkUpdateService.AvailableUpdate();
+
+            if (latestAvailable == null || latestAvailable.Hash != message.UpdatePackage.Hash)
+            {
+                throw new ApplicationException("Unknown or invalid update specified");
```

Comment:
We should update the code so it just uses the version sent from UI and ignores the hash and url.